src/components/shared/Map/index.tsx

```diff
@@ -1,35 +1,57 @@
 import 'leaflet/dist/leaflet.css';
+import { useEffect, useState } from 'react';
 import { MapContainer, TileLayer } from 'react-leaflet';
 
+import getRecent from '@/services/DataService';
+import { getLocation, getLocations } from '@/services/StationsService';
+import { type SensorData } from '@/types/models/SensorData';
+import {
+  type SensorLocation,
+  type SensorLocationData
+} from '@/types/models/SensorLocation';
+
 import MapPointer from './MapPointer';
 
 export default function Map() {
+  const [locations, setLocations] = useState<[SensorLocation]>([]);
+  const [sensorData, setSensorData] = useState<[SensorData]>([]);
+  useEffect(() => {
+    void getLocations().then(async res => {
+      setLocations(res);
+      const data = await getRecent();
+      setSensorData(data);
+    });
+    return () => {};
+  }, []);
+
+  let i = 0;
+  const sensorLocationData: [SensorLocationData] = [];
+  sensorData.forEach((value: SensorData) => {
+    const sensorLocation: SensorLocation = getLocation(value.code, locations);
+    sensorLocationData.push({
+      data: value,
+      location: sensorLocation,
+      key: i
+    });
+    i += 1;
+  });
```

Comment:
```suggestion
  const sensorLocationData = useMemo(
    () =>
      sensorData.map((value: SensorData, idx) => ({
        data: value,
        location: getLocation(value.code, locations),
        key: idx
      })),
    [locations, sensorData]
  );
```
For more complex calculations use `useMemo` hook. It will look much cleaner and recalculate values only when specific dependencies change (not on every component render). In your case, component re-renders when ether `locations` or `sensorData` changes, so `useMemo` will be recalculated on each re-render but it still makes code more ogranized.